app/lib/shared/handler_helpers.dart

```diff
@@ -19,22 +19,28 @@ import 'handlers.dart';
 import 'markdown.dart';
 import 'utils.dart' show fileAnIssueContent;
 
-Future runHandler(Logger logger, shelf.Handler handler,
-    {bool sanitize = false}) {
+Future<void> runHandler(
+  Logger logger,
+  shelf.Handler handler, {
+  bool sanitize = false,
+  shelf.Handler cronJobHandler,
+}) async {
   handler = _uriValidationRequestWrapper(handler);
   handler = _userAuthParsingWrapper(handler);
   if (sanitize) {
     handler = _sanitizeRequestWrapper(handler);
   }
   handler = _redirectToHttpsWrapper(handler);
   handler = _logRequestWrapper(logger, handler);
-  return runAppEngine((HttpRequest request) => _handleRequest(request, handler),
-      shared: true);
+  await runAppEngine((HttpRequest request) {
+    if (cronJobHandler != null && isCronJobRequest(request)) {
+      shelf_io.handleRequest(request, cronJobHandler);
```

Comment:
Why not put this inside the `appHandler` in `lib/frontend/handlers.dart`? I know that technically this is not an app-related handler, but I would probably look there first to see where this is going to be routed.

Reply:
The `isCronJobRequest` operates on `HttpRequest` and not `shelf.Request`...

We could just reimplement it using `shelf.Request`... I'll agree this all feels a little stitched together..